client/web/src/components/exit-node-selector.tsx

```diff
@@ -131,47 +176,381 @@ export default function ExitNodeSelector({
           </button>
         )}
       </div>
-      {open && (
-        <div className="absolute ml-1.5 -mt-3 w-full max-w-md py-1 bg-white rounded-lg shadow">
-          <div className="w-full px-4 py-2 flex items-center gap-2.5">
-            <Search />
-            <input
-              className="flex-1 leading-snug"
-              placeholder="Search exit nodes…"
-            />
-          </div>
-          <DropdownSection
-            items={[noExitNode, runAsExitNode]}
-            selected={selected}
-            onSelect={handleSelect}
-          />
-        </div>
-      )}
-    </>
+    </Popover>
   )
 }
 
-function DropdownSection({
-  items,
+function toSelectedExitNode(data: NodeData): ExitNode {
+  if (data.AdvertiseExitNode) {
+    return runAsExitNode
+  }
+  if (data.ExitNodeStatus) {
+    // TODO(sonia): also use online status
+    const node = { ...data.ExitNodeStatus }
+    if (node.Location) {
+      node.Name = `${node.Location.Country}: ${node.Location.City}`
+    }
+    node.Name = trimDNSSuffix(node.Name, data.TailnetName)
```

Comment:
nit: shouldn't likely matter, but maybe put this in an else of the above `if (node.Location)`, since those options are mutually exclusive?